paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/PaymentMethodMetadata.kt

```diff
@@ -70,19 +86,38 @@ internal data class PaymentMethodMetadata(
         }
     }
 
-    fun supportedPaymentMethodForCode(code: String): SupportedPaymentMethod? {
+    fun supportedPaymentMethodForCode(
+        code: String,
+        context: Context,
+        paymentMethodCreateParams: PaymentMethodCreateParams? = null,
+        paymentMethodExtraParams: PaymentMethodExtraParams? = null,
+    ): SupportedPaymentMethod? {
         val definition = supportedPaymentMethodDefinitions().firstOrNull { it.type.code == code } ?: return null
         val sharedDataSpec = sharedDataSpecs.firstOrNull { it.type == code } ?: return null
-        return definition.supportedPaymentMethod(this, sharedDataSpec)
+        return definition.supportedPaymentMethod(
+            metadata = this,
+            sharedDataSpec = sharedDataSpec,
+            transformSpecToElements = transformSpecToElements(
+                context = context,
```

Comment:
Don't think we avoid passing `context` here without making fundamental changes to how `CardNumberController` works right?